app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -419,5 +419,16 @@ def request_url
     def logger
       ::Foreman::Logging.logger('katello/registry_proxy')
     end
+
+    def route_name
+      Engine.routes.router.recognize(request) do |_, params|
+        break params[:action] if params[:action]
```

Comment:
I'm not sure what this does, does it bail out if `params[:action]` is defined?

Reply:
This is a pattern for finding the name of the route being hit in rails. I had to goggle to figure this out but apparently there isn't an easy way to get the route. When the action is found, the method returns (breaks loop and falls through).